modules/storage/cosmos-db/main.bicep

```diff
@@ -1,99 +1,353 @@
-@description('The default consistency level of the Cosmos DB account.')
-param defaultConsistencyLevel string = 'Session'
+@description('MongoDB server version. Required for mongodb API type Cosmos DB account')
+@allowed([ '3.2', '3.6', '4.0', '4.2' ])
+param serverVersion string = '4.2'
 
-@description('Enable system managed failover for regions')
-param systemManagedFailover bool = true
+@description('List of CORS rules.')
+param cors array = []
 
-@description('array of region objects or regions: [region: string]')
-param secondaryLocations array = []
+@description('Disable write operations on metadata resources (databases, containers, throughput) via account keys.')
+param disableKeyBasedMetadataWriteAccess bool = false
 
-@description('Multi-region writes capability allows you to take advantage of the provisioned throughput for your databases and containers across the globe.')
-param enableMultipleWriteLocations bool = true
+@description('Whether requests from public network allowed.')
+@allowed([ 'Enabled', 'Disabled' ])
+param publicNetworkAccess string = 'Enabled'
 
-@description('Enable Cassandra Backend.')
-param enableCassandra bool = false
+@description('Flag to indicate whether to enable/disable Virtual Network ACL rules.')
+param isVirtualNetworkFilterEnabled bool = false
 
-@description('Enable Serverless for consumption-based usage.')
-param enableServerless bool = false
+@allowed([ 'AzureServices', 'None' ])
+@description('Indicates what services are allowed to bypass firewall checks.')
+param networkAclBypass string = 'None'
 
-@description('Toggle to enable or disable zone redudance.')
-param isZoneRedundant bool = false
+@description('List of IpRules to be allowed. A single IPv4 address or a single IPv4 address range in CIDR format.')
+param ipRules array = []
 
-var consistencyPolicy = {
-  Eventual: {
-    defaultConsistencyLevel: 'Eventual'
-  }
-  ConsistentPrefix: {
-    defaultConsistencyLevel: 'ConsistentPrefix'
-  }
-  Session: {
-    defaultConsistencyLevel: 'Session'
-  }
-  BoundedStaleness: {
-    defaultConsistencyLevel: 'BoundedStaleness'
-    maxStalenessPrefix: maxStalenessPrefix
-    maxIntervalInSeconds: maxIntervalInSeconds
-  }
-  Strong: {
-    defaultConsistencyLevel: 'Strong'
-  }
+@description('The list of virtual network ACL rules, "isVirtualNetworkFilterEnabled" must be set to "true". format: {id: string, ignoreMissingVNetServiceEndpoint: bool}')
+param virtualNetworkRules array = []
+
+@description('An array that contains the Resource Ids for Network Acl Bypass.')
+param networkAclBypassResourceIds array = []
+
+@description('Non-default extra capabilities.')
+@allowed([ 'DisableRateLimitingResponses', 'EnableAggregationPipeline', 'EnableMongo16MBDocumentSupport', 'EnableMongoRetryableWrites', 'EnableMongoRoleBasedAccessControl', 'EnableUniqueCompoundNestedDocs', 'mongoEnableDocLevelTTL' ])
+param capabilities array = []
```

Comment:
I feel like there is a longer list of capabilities... Somewhere i started to track them down, but they arent part of the official schema. Let's not include @allowed here, 

@jeskew, FYI literally the case I think we were talking about with Cosmos.

Reply:
Deleted `@allowed` from capabilities.